osu.Game/Screens/Select/PlaySongSelect.cs

```diff
@@ -101,7 +102,7 @@ protected override bool OnStart()
 
                 if (mods.All(m => m.GetType() != autoType))
                 {
-                    Mods.Value = mods.Append(auto).ToArray();
+                    Mods.Value = mods.Where(m => !m.IncompatibleMods.Contains(typeof(ModAutoplay))).Append(auto).ToArray();
```

Comment:
Oh, and also there's this - if a mod doesn't specify the base `ModAutoplay` but a deriving class in `SelectedMods`, then I'm pretty sure this won't work.

Reply:
Why was this resolved?